service/history/execution/mutable_state_task_generator.go

```diff
@@ -145,10 +136,11 @@ func (r *mutableStateTaskGeneratorImpl) GenerateWorkflowStartTasks(
 
 	executionInfo := r.mutableState.GetExecutionInfo()
 	startVersion := startEvent.GetVersion()
+	startTimestamp := time.Unix(0, startEvent.GetTimestamp())
 
 	workflowTimeoutDuration := time.Duration(executionInfo.WorkflowTimeout) * time.Second
 	workflowTimeoutDuration = workflowTimeoutDuration + firstDecisionDelayDuration
-	workflowTimeoutTimestamp := now.Add(workflowTimeoutDuration)
+	workflowTimeoutTimestamp := startTimestamp.Add(workflowTimeoutDuration)
```

Comment:
Looks like we still need to pass in a timestamp here since workflow reset needs to use reset time instead of start event time for calculating the timeout timestamp. 
Will revert the changes for this function to preserve the now parameter.

Reply:
How about change the parameter name to something else. It’s not really now. Also a comment would help understand why we don’t use start event timestamp

Reply:
Eg: timeoutStartTime

Reply:
sure. I renamed it to `startTime` and added a comments says for reset the workflow start time should be the reset time in order to re-calculated the timeout timestamp.